packages/common/src/util/http.js

```diff
@@ -245,6 +245,10 @@ async function readResponseBody(response, parseAs) {
       return undefined;
     }
 
+    if (response.statusCode === 204) {
```

Comment:
I have a feeling we need to remove this (and the associated changeset).  `main` has had a bunch of changes in this area and this branch should be stable without this diff